pkg/operator/configobservation/auth/external_oidc.go

```diff
@@ -0,0 +1,159 @@
+		}
+
+		// empty source name/namespace effectively deletes target configmap
+		if err := syncConfigMap(genericListers.ResourceSyncer(), "", "", recorder); err != nil {
+			return existingConfig, []error{err}
+		}
+
+		return nil, nil
+	}
+
+	// auth type is OIDC
+
+	sourceAuthConfig, err := listers.ConfigMapLister().ConfigMaps(SourceAuthConfigCMNamespace).Get(AuthConfigCMName)
+	if errors.IsNotFound(err) {
+		klog.Warningf("configmap %s/%s not found; skipping configuration of OIDC", SourceAuthConfigCMNamespace, AuthConfigCMName)
+		return existingConfig, nil
+
+	} else if err != nil {
+		return existingConfig, []error{fmt.Errorf("failed to get configmap %s/%s: %v", SourceAuthConfigCMNamespace, AuthConfigCMName, err)}
+	}
+
+	if err := validateAuthConfigCM(sourceAuthConfig); err != nil {
+		return existingConfig, []error{fmt.Errorf("configmap %s/%s is invalid: %v", SourceAuthConfigCMNamespace, AuthConfigCMName, err)}
+	}
+
+	targetAuthConfig, err := listers.ConfigMapLister().ConfigMaps(operatorclient.TargetNamespace).Get(AuthConfigCMName)
+	if err != nil && !errors.IsNotFound(err) {
+		return existingConfig, []error{err}
+	}
+
+	if targetAuthConfig == nil || !equality.Semantic.DeepEqual(targetAuthConfig.Data, sourceAuthConfig.Data) {
+		if err := syncConfigMap(genericListers.ResourceSyncer(), sourceAuthConfig.Name, sourceAuthConfig.Namespace, recorder); err != nil {
+			return existingConfig, []error{err}
+		}
+	}
+
+	observedConfig := make(map[string]interface{})
+	if err := unstructured.SetNestedField(observedConfig, []interface{}{staticAuthConfigPath}, apiServerArgumentsPath, argAuthConfig); err != nil {
+		return existingConfig, []error{err}
+	}
+
+	return observedConfig, nil
+}
+
+func syncConfigMap(syncer resourcesynccontroller.ResourceSyncer, sourceName, sourceNamespace string, recorder events.Recorder) error {
+	if err := syncer.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: AuthConfigCMName},
+		resourcesynccontroller.ResourceLocation{Namespace: sourceNamespace, Name: sourceName},
+	); err != nil {
+		return err
+	}
+
+	if len(sourceName) == 0 {
+		recorder.Eventf("ObserveExternalOIDC", "deleted configmap %s/%s", operatorclient.TargetNamespace, AuthConfigCMName)
+	} else {
+		recorder.Eventf("ObserveExternalOIDC", "synced configmap %s/%s to %s/%s", sourceNamespace, sourceName, operatorclient.TargetNamespace, AuthConfigCMName)
+	}
+
+	return nil
+}
+
+func validateAuthConfigCM(cm *corev1.ConfigMap) error {
+	if cm == nil {
+		return fmt.Errorf("configmap is nil")
+	}
+
+	authConfigRaw, ok := cm.Data["auth-config.json"]
+	if !ok {
+		return fmt.Errorf("missing required 'auth-config.json' key")
+	}
+
+	if len(authConfigRaw) == 0 {
+		return fmt.Errorf("value of key 'auth-config.json' is empty")
+	}
+
+	var authConfig apiserver.AuthenticationConfiguration
```

Comment:
https://pkg.go.dev/k8s.io/apiserver/pkg/apis/apiserver/load only contains `Load*()` funcs for the `AuthorizationConfiguration` configuration type, instead of `AuthenticationConfiguration` which we need here.

Reply:
Oh, sorry, missed that. Then you'll need to set up the codec yourself here.